ruby/ext/google/protobuf_c/storage.c

```diff
@@ -725,20 +729,20 @@ static void slot_clear_hasbit(MessageLayout* layout,
 static bool slot_is_hasbit_set(MessageLayout* layout,
                             const void* storage,
                             const upb_fielddef* field) {
-  assert(field_contains_hasbit(layout, field));
   size_t hasbit = layout->fields[upb_fielddef_index(field)].hasbit;
+  assert(field_contains_hasbit(layout, field));
```

Comment:
Do you need to move this assert?

Reply:
Yes, because it is a statement and cannot go before the declaration of `hasbit`.

This worked previously because we had not been testing with asserts enabled.